### PR TITLE
Update hdf5 daos vol/vol tests to latest commits

### PR DESCRIPTION
**Pull request by mjean308**
Signed-off-by: Maureen Jean <maureen.jean@intel.com>

### CODE REVIEW

**Comment by mjean308** [comment]
This rpm is currently running with daos PR-3413

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,8 @@
 NAME          := hdf5-vol-daos
 SRC_EXT       := gz
 TEST_PACKAGES := daos-devel $(NAME) $(NAME)-mpich $(NAME)-openmpi3 $(NAME)-mpich-devel $(NAME)-openmpi3-devel $(NAME)-mpich-tests $(NAME)-openmpi3-tests
-SOURCE_COMMIT := b324b90d4554e211976174b9a3acdba61c184248
-TEST_COMMIT   := 31660ab19352049cf59f98b1fba498b3e93dade5
+SOURCE_COMMIT := 8d419310976bcd890521ebf72557325f34a8af56
+TEST_COMMIT   := d59811a39cd5053887f22cdabd1c08e6d9c539db
 SOURCE        = $(SOURCE_COMMIT).tar.$(SRC_EXT) 
 PATCHES       = $(TEST_COMMIT).tar.$(SRC_EXT)
```
Comment (brianjmurrell) [inline Makefile:7]: Are these two values used anywhere?  Should they be removed?  `$(PATCHES)` isn't really any patches yes?

Comment (mjean308) [reply Makefile]: No they are not used and I removed them

Comment (mjean308) [reply Makefile:7]: They are needed;  They override the SOURCE and PATCHES in Makefile_packaging.mk.  After I removed them the build failed with the following error


[2020-09-11T12:06:23.928Z] + rm -rf artifacts/centos7/

[2020-09-11T12:06:23.928Z] + mkdir -p artifacts/centos7/

[2020-09-11T12:06:23.928Z] + make CHROOT_NAME=epel-7-x86_64 chrootbuild

[2020-09-11T12:06:24.499Z] make: *** No rule to make target '%{source_commit}.tar.gz', needed by '_topdir/SOURCES/%{source_commit}.tar.gz'.  Stop.

Comment (brianjmurrell) [reply Makefile:7]: That seems like a deficiency in the packaging.  Probably the macros are not being defined in the spectool calls that assign the `SOURCES` and `PATCHES` values in the `packaging/Makefile_packaging.mk`.

**Comment by mjean308** [comment]
The daos PR completed successfully with DAOS-5469 and DAOS-5496 verified as still failing.   ie this update did not fix either test failure.

Mohamad, can you verify if h5_test_testhdf5 (DAOS-5469) and test h5_partest_t_bigio (DAOS-5496) are expected to fail with this update?

**Review by mchaarawi — DISMISSED** [review]

**Comment by brianjmurrell** [comment]
> The daos PR completed successfully with DAOS-5469 and DAOS-5496 skipped and verified as still failing. ie this update did not fix either test failure.
> 
> Mohamad, can you verify if h5_test_testhdf5 (DAOS-5469) and test h5_partest_t_bigio (DAOS-5496) are expected to fail with this update?

Was there an answer to this question?  Was this PR expected to resolve those issues and didn't?

**Comment by mjean308** [comment]
Both tests are still failing.  DAOS-5469 footprint changed but is still failing and this test will be blocked until we release HDF5 1.12.1.  Ticket has been updated to note the change and has been blocked.   DAOS-5496 still fails as well.      However, Mohamad would like to go ahead and release this update.

**Comment by mchaarawi** [comment]
> Both tests are still failing. DAOS-5469 footprint changed but is still failing and this test will be blocked until we release HDF5 1.12.1. Ticket has been updated to note the change and has been blocked. DAOS-5496 still fails as well. However, Mohamad would like to go ahead and release this update.

right both tests still have issues.

File: Makefile (hunk 1 above)
Comment (brianjmurrell) [reply Makefile:7]: @mjean308 What about this?  It seems like it should be a pretty simple fix.  I really should have caught it on the PR it landed in. You should be able to just add `$(BUILD_DEFINES)` to the `spectool` arguments.

**Review by mchaarawi — DISMISSED** [review]

**Comment by mjean308** [comment]
Running daos-stack/daos#3812 to verify rpm changes run in Functional HW tests

File: hdf5-vol-daos.spec [hunk 2]
```diff
@@ -1,7 +1,7 @@
 %global with_mpich 1
 %global with_openmpi3 1
 
-%global daos_major 0
+%global daos_major 1
```
Comment (brianjmurrell) [inline hdf5-vol-daos.spec:4]: The libdaos major version bump has not landed to master.
```suggestion
%global daos_major 0
```

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
Similar to the hdf5 review comments, this should be refactored to base the updated commits as a `Patch:` in the specfile on top of the [most recent upstream release](https://github.com/HDFGroup/vol-daos/releases/tag/v1.0.0rc1).

The tests commit should also be a `Patch:` in the specfile.

**Review by mchaarawi — APPROVED** [review]

**Comment by mjean308** [comment]
PR-3812 is currently in progress to test ior-hpc; hdf5-vol-daos and hdf5 in daos CI

**Review by brianjmurrell — APPROVED** [review]
I will +1 this time, but the next update to this package needs to move to the standard release+patch methodology please.

**Comment by brianjmurrell** [comment]
> PR-3812 is currently in progress to test ior-hpc; hdf5-vol-daos and hdf5 in daos CI

daos-stack/daos#3812 seems to have [failed DaosVol.test_daos_vol](https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-3812/13/testReport/junit/(root)/DaosVol/Test___Functional_Hardware_Small___06___io_daos_vol_py_DaosVol_test_daos_vol_container_test6_dfuse_hosts_mpich_pool_servers_f6cd/).  Should this land with that failure?  Won't that introduce a new failure in master and PRs?

**Comment by mjean308** [comment]
All daos vol tests (unit tests and ior) are currently disabled in CI. The new vol unit test failure has been bugged with DAOS-6076.
When the vol tests are re-enabled this new failure will be skipped as well as the existing bugs DAOS-5469 and DAOS-5496.

**Comment by mjean308** [comment]
The hdf5 vol unit tests have passed with the following tests skipped:
        h5_partest_t_shapesame - DAOS-5831
        h5_test_testhdf5 - DAOS-5469
        h5_partest_testphdf5 - DAOS-6076
        h5_partest_t_bigio DAOS-5496
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-3812/18/pipeline/489

ior with hdf5 vol have also passed CI
https://build.hpdd.intel.com/blue/organizations/jenkins/daos-stack%2Fdaos/detail/PR-3812/13/tests 

This PR is ready to land

**Review by mchaarawi — APPROVED** [review]

File: hdf5-vol-daos.spec [hunk 3]
```diff
@@ -197,8 +197,11 @@ done
 %endif
 
 %changelog
+* Thu Nov  5 2020 Maureen Jean <maureen.jean@intel.com> - 0.1-3.g9afaf5907
+- Update hdf5-vol and vol-test to g1d2a3acb4
```
Comment (brianjmurrell) [inline hdf5-vol-daos.spec:201]: This doesn't seem to reflect the actual git hashes used and is only referring to a single hash when it's two different repos.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

**Review by brianjmurrell — APPROVED** [review]

**Pull request merged**